homeassistant/components/hydrawise/config_flow.py

```diff
@@ -91,15 +130,26 @@ def _show_form(self, error_type: str | None = None) -> ConfigFlowResult:
             errors["base"] = error_type
         return self.async_show_form(
             step_id="user",
-            data_schema=vol.Schema({vol.Required(CONF_API_KEY): str}),
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+            ),
             errors=errors,
         )
 
+    async def async_step_reauth(
+        self, user_input: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth after updating config to username/password."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_user()
+
     async def async_step_import(self, import_data: dict[str, Any]) -> ConfigFlowResult:
         """Import data from YAML."""
         try:
-            result = await self._create_entry(
-                import_data.get(CONF_API_KEY, ""),
+            result = await self._create_or_update_entry(
+                api_key=import_data.get(CONF_API_KEY, ""),
```

Comment:
It's not correct to use `dict.get` here since configuring an API token is mandatory in the YAML schema.
```suggestion
                api_key=import_data[CONF_API_KEY],
```